R/utilities.R

```diff
@@ -1148,3 +1148,185 @@ ToNumeric <- function(x){
+      warning(sprintf(
+        'Embedding %s has more than 2 coordinates, taking only the first 2',
+        embedding
+      ))
+      df <- df[, 1:2]
+    }
+    colnames(df) <- c("x", "y")
+    df <- data.frame(cellId = rownames(df), df)
+    fname <- file.path(
+      dir,
+      sprintf("%s.coords.tsv", embedding)
+    )
+    write.table(df[order, ], sep="\t", file=fname, quote = FALSE, row.names = FALSE)
+    conf <- sprintf(
+      '{"file": "%s.coords.tsv", "shortLabel": "Seurat %1$s"}',
+      embedding
+    )
+    embeddings.conf <- c(embeddings.conf, conf)
+  }
+
+  # Export metadata
+  df <- data.frame(row.names = rownames(object[[]]))
+  for (field in meta.fields) {
+    if (field == ".ident") {
+      df$Cluster <- Idents(object = object)
+      enum.fields <- c(enum.fields, "Cluster")
+      cluster.field <- "Cluster"
+    } else {
+      name <- meta.fields.names[[field]]
+      if (is.null(name)) {
+        name <- field
+      }
+      df[[name]] <- object[[field]][, 1]
+      if (!is.numeric(df[[name]])) {
+        enum.fields <- c(enum.fields, name)
+      }
+    }
+  }
+  df <- data.frame(Cell=rownames(df), df)
+  write.table(df[order, ], sep="\t", file=file.path(dir, "meta.tsv"), quote = FALSE, row.names=FALSE)
+
+  # Export markers
+  markers.string <- ''
+  if (!is.null(markers.file)) {
+    file.copy(markers.file, file.path(dir, "markers.tsv"))
+    markers.string <- 'markers = [{"file": "markers.tsv", "shortLabel": "Seurat Cluster Markers"}]'
+  }
+
+  config <- 'name="%s"
+shortLabel="%1$s"
+exprMatrix="exprMatrix.tsv.gz"
+#tags = ["10x", "smartseq2"]
+meta="meta.tsv"
+# possible values: "gencode-human", "gencode-mouse", "symbol" or "auto"
+geneIdType="auto"
+clusterField="%s"
+labelField="%2$s"
+enumFields=%s
+%s
+coords=%s'
+  enum.string <- paste0(
+    "[",
+    paste(paste0('"', enum.fields, '"'), collapse = ", "),
+    "]"
+  )
+  coords.string <- paste0(
+    "[",
+    paste(embeddings.conf, collapse = ",\n"),
+    "]"
+  )
+  config <- sprintf(
+    config,
+    dataset.name,
+    cluster.field,
+    enum.string,
+    markers.string,
+    coords.string
+  )
+  cat(config, file=file.path(dir, "cellbrowser.conf"))
+  message("Prepared cellbrowser directory ", dir)
+
+  if (!is.null(cb.dir) && require(reticulate)) {
+    library(reticulate)
+    if (!py_module_available("cellbrowser")) {
+      py_install("cellbrowser") # see https://rstudio.github.io/reticulate/articles/python_packages.html
+    }
+
+    if (!is.null(port)) {
+      port <- as.integer(port)
+    }
+    cb <- import("cellbrowser") # see https://github.com/rstudio/reticulate/issues/73
+    cb$cellbrowser$build(dir, cb.dir)
+    if (!is.null(port)) {
+      cb$cellbrowser$stop()
+      cb$cellbrowser$serve(cb.dir, port)
+      utils::browseURL(sprintf("http://localhost:%d", port))
+    }
+  }
+}
+
+#' Stop Cellbrowser web server
+#'
+#' @export
+#'
+#' @examples
+#' StopCellbrowser()
+#'
+StopCellbrowser <- function() {
+    library(reticulate)
```

Comment:
Similarly, please use the `importFrom` tag within the roxygen documentation instead of `library`